apps/indexer/src/api/services/proposals/proposals.ts

```diff
@@ -23,24 +24,32 @@ export class ProposalsService {
     limit = 10,
     orderDirection = "desc",
     status,
+    status_in,
     fromDate,
   }: ProposalsRequest): Promise<DBProposal[]> {
     const proposals = await this.proposalsRepo.getProposals(
       skip,
       limit,
       orderDirection,
       status,
+      status_in,
       fromDate,
     );
 
     for (const proposal of proposals) {
       proposal.status = await this.daoClient.getProposalStatus(proposal);
     }
 
+    // Filter by status or status_in if provided
     if (status) {
       // filtering proposals marked as "PENDING" on the database,
       // but representing different statuses onchain
       return proposals.filter((proposal) => proposal.status === status);
+    } else if (status_in && status_in.length > 0) {
+      // filtering proposals by multiple statuses
+      return proposals.filter((proposal) =>
+        status_in.includes(proposal.status),
+      );
     }
 
```

Comment:
Inefficient query logic: The code fetches all proposals from database, then calls getProposalStatus() for each one, then filters in memory. This means if you have 1000 PENDING proposals but only want ACTIVE ones, you're making 1000 unnecessary API calls. Consider moving the status resolution logic to the database layer or implementing smarter filtering to avoid N+1 performance issues.